### PR TITLE
Add technical note re: AVPlayer recording (iOS)

### PR DESCRIPTION
**Pull request by pavel-kroh-smartlook**
There are still inquiries and bug reports about recording screens with AVPlayer (which does not work). 

This technical note describes the symptoms, causes and a possible workaround for some use-cases to be shared with customers who run into it.

### CODE REVIEW

File: src/cookbook/ios-avplayer-recording.mdx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import { Title, TextBlock } from 'components'
+
+<Title>iOS technical note: AVPlayer recording</Title>
+
+<TextBlock kind="important">{`
+This guide applies to all frameworks that create or generate iOS applications.
+`}</TextBlock>
+
+Support for recording video from `AVPlayerLayer` or from camera preview via `AVCaptureVideoPreviewLayer` is not currently available.
```
Comment (SenNeonoveNoci) [inline src/cookbook/ios-avplayer-recording.mdx:16]: Remove second "from":
"from `AVPlayerLayer` or from camera" -> "from `AVPlayerLayer` or camera"

File: src/cookbook/ios-avplayer-recording.mdx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Symptoms
+
+- When user plays video with `AVPlayer` in layer or in `AVPlayerViewController`, <b>the video is not recorded and the background (video content) is black</b>.
+
+- All camera previews are black (without content).
+
+## Technical reasons
+
+Both layers uses low level API that serves data to this layers via pixel buffers. Graphic data from `AVFoundation` are processed directly on device GPU with hardware support, not in `Core Graphics` framework like other GUI parts. For us are these low level layer inaccessible.
```
Comment (SenNeonoveNoci) [inline src/cookbook/ios-avplayer-recording.mdx:26]: "uses" -> "use"
"low level" -> "low-level" 
"this layers" -> "these layers"
"on device" -> "on-device"
"For us are these low level layer inaccessible." -> "For us are these low-level layers inaccessible."

File: src/cookbook/ios-avplayer-recording.mdx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+<TextBlock kind="note">{`
+\`AVPlayer\` recording failure is obviously an architectural problem that has not been solved by Apple for a long time. Their own \`ReplayKit\` framework behaves exactly the same, and Apple itself points this out in the documentation.\n
+In general, using the camera preview does not mean that recording is not working.
```
Comment (SenNeonoveNoci) [inline src/cookbook/ios-avplayer-recording.mdx:30]: "recording" -> "the recording"

File: src/cookbook/ios-avplayer-recording.mdx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+<TextBlock kind="note">{`
+\`AVPlayer\` recording failure is obviously an architectural problem that has not been solved by Apple for a long time. Their own \`ReplayKit\` framework behaves exactly the same, and Apple itself points this out in the documentation.\n
+In general, using the camera preview does not mean that recording is not working. 
+For example, in \`ARKit\` we do not have problems because internally uses \`MetalKit\`. `}</TextBlock>
+
+## Recommendations
+
+### Camera preview
+Do not use `AVCaptureVideoPreviewLayer` to render the data from device camera.
```
Comment (SenNeonoveNoci) [inline src/cookbook/ios-avplayer-recording.mdx:36]: "device" -> "the device"

File: src/cookbook/ios-avplayer-recording.mdx (hunk 2 above)
Comment (pavel-kroh-smartlook) [reply src/cookbook/ios-avplayer-recording.mdx]: Fixed.

File: src/cookbook/ios-avplayer-recording.mdx (hunk 4 above)
Comment (pavel-kroh-smartlook) [reply src/cookbook/ios-avplayer-recording.mdx]: Fixed.

**Review by SenNeonoveNoci — APPROVED** [review]

**Pull request merged**